pkgs/by-name/li/libwacom/package.nix

```
@@ -58,37 +51,33 @@ stdenv.mkDerivation (finalAttrs: {
     libgudev
   ];
 
-  # Tests are in the `tests` pass-through derivation.
-  # See https://github.com/NixOS/nixpkgs/issues/328140
-  doCheck = false;
   mesonFlags = [
-    testsDisabled
-    "--sysconfdir=/etc"
+    (lib.mesonEnable "tests" withTests)
+    (lib.mesonOption "sysconfdir" "/etc")
   ];
 
-  passthru = {
-    tests = finalAttrs.finalPackage.overrideAttrs (prevAttrs: {
-      doCheck = true;
+  doCheck = withTests;
 
-      mesonFlags = map replaceDisabledWithEnabled prevAttrs.mesonFlags;
+  nativeCheckInputs = lib.optionals withTests [
+    valgrind
+    (python3.withPackages (ps: [
+      ps.libevdev
+      ps.pytest
+      ps.pyudev
+    ]))
+  ];
 
-      nativeCheckInputs = [
-        valgrind
-        (python3.withPackages (ps: [
-          ps.libevdev
-          ps.pytest
-          ps.pyudev
-        ]))
-      ];
-    });
+  passthru = {
+    # finalAttrs.finalPackage does not have .override
+    tests = libwacom.override { withTests = true; };
```

Let's keep this finalAttrs.finalPackage.overrideAttrs so that libwacom-surface has the right `.tests` derivation.

```suggestion
    tests = finalAttrs.finalPackage.overrideAttrs (prevAttrs: { doCheck = true; });
```

---

As mentioned, this is undesirable because it makes our code depend on `doCheck` which is an implementation detail of `mkDerivation`.

OTOH, we also need to pass through `libwacom` to itself which is a bit of an odd pattern and comes with some caveats.

Additionally, this isn't really something we need to expose externally as that flag itself is an implementation detail for producing a passthru attr.

---

What I'm saying is that `libwacom-surface` is defined as `libwacom.overrideAttrs`.

https://github.com/NixOS/nixpkgs/blob/b8e345cd4a8bd35b08858a87a12326a0099557a8/pkgs/by-name/li/libwacom-surface/package.nix

That means `nix-build -A libwacom-surface.tests` is going to run tests against the base, not against the patched version.

---

Yeah, that's simpler and breaks fewer surprises. Please pull.